crates/bevy_render/src/view/window/mod.rs

```diff
@@ -34,6 +36,20 @@ pub struct WindowRenderPlugin;
 impl Plugin for WindowRenderPlugin {
     fn build(&self, app: &mut App) {
         app.add_plugins(ScreenshotPlugin);
+        #[cfg(all(
+            feature = "bevy_winit",
+            feature = "multi-threaded",
+            not(target_arch = "wasm32")
+        ))]
+        {
+            // TODO: This is added despite not checking if PipelinedRenderingPlugin is added or not,
+            // but we can't check right now (since it's added after this plugin)
+            if app.is_plugin_added::<bevy_winit::WinitPlugin>() {
+                app.add_plugins(winit::WinitWindowRenderPlugin);
+            } else {
+                bevy_utils::tracing::warn!("Winit feature was enabled but couldn't detect WinitPlugin. Are you sure you loaded this after WinitPlugin?");
+            }
+        }
```

Comment:
This potentially hard-couples bevy_render and bevy_winit. I'm not sure if we want to do this.

Reply:
Take a look at this: https://github.com/Friz64/bevy/commit/0561754959e8051acf448a64a7ac982177a44def

It also works as a fix, and avoids the hard-coupling to `bevy_winit`. I'm not convinced that dynamic dispatch is the most elegant solution here, but it works.